src/encoded/tests/test_schemas.py

```diff
@@ -220,3 +219,48 @@ def test_unique_keys_are_identifying_properties(testapp: TestApp) -> None:
 def has_identifying_property(schema: Dict[str, Any], property_name: str) -> bool:
     """Return True if schema has property_name property."""
     return property_name in schema_utils.get_identifying_properties(schema)
+
+
+def test_submittable(testapp, registry):
+    expected_props = [
+        'a260_a280_ratio', 'analyte_preparation', 'components', 'concentration',
+        'molecule', 'protocols', 'ribosomal_rna_ratio', 'rna_integrity_number',
+        'rna_integrity_number_instrument', 'sample_quanitity', 'samples',
+        'submitted_id', 'tags', 'volume', 'weight',
+     ]
+    expected_req = ["components", "molecule", "samples", "submitted_id"]
+    expected_dep_req = ['rna_integrity_number', 'rna_integrity_number_instrument']
+    schema_name = 'analyte.json'
+    loaded_schema = load_schema('encoded:schemas/%s' % schema_name)
+    all_propnames = schema_utils.get_properties(loaded_schema).keys()
+    test_uri = f'/submission-schemas/{schema_name}'
+    res = testapp.get(test_uri).json
+    sub_props = schema_utils.get_properties(res)
+    assert len(sub_props) == len(expected_props)
+    non_sub_prop_cnt = 0
+    for prop in all_propnames:
+        if prop in expected_props:
+            assert prop in sub_props
+            if prop in expected_req:
+                assert 'is_required' in sub_props[prop]
+            if prop in expected_dep_req:
+                assert 'also_requires' in sub_props[prop]
+        else:
+            non_sub_prop_cnt += 1
+    assert len(sub_props) + non_sub_prop_cnt == len(all_propnames)
+
+
+def test_submittables(testapp, registry):
+    test_uri = '/submission-schemas/'
+    expected_items = [
+        'AlignedReads', 'Analyte', 'AnalytePreparation', 'CellCulture',
+        'CellCultureMixture', 'CellCultureSample', 'CellLine', 'CellSample',
+        'DeathCircumstances', 'Demographic', 'Diagnosis', 'Donor', 'Exposure',
+        'FileSet', 'Histology', 'Library', 'LibraryPreparation', 'MedicalHistory',
+        'MolecularTest', 'PreparationKit', 'SamplePreparation', 'Sequencing',
+        'Software', 'Therapeutic', 'Tissue', 'TissueCollection', 'TissueSample',
+        'Treatment', 'UnalignedReads', 'VariantCalls']
```

Comment:
Consider replacing with [this function](https://github.com/smaht-dac/smaht-portal/blob/main/src/encoded/tests/utils.py#L254-L264) to make easier to maintain.

Reply:
done